GliaWidgets/Sources/QuickLookBased/QuickLookBased.Implementation.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import UIKit
+import QuickLook
+
+extension QuickLookBased.ThumbnailGenerator {
+    static func create(uiScreen: UIKitBased.UIScreen) -> Self {
+        if #available(iOS 13.0, *) {
+            let generator = QLThumbnailGenerator.shared
+            return .init(
+                generateBestRepresentation: { size, url, completion in
+                    let scale = uiScreen.scale()
+                    let newSize = CGSize(width: size.width * scale, height: size.height * scale)
+                    let request = QLThumbnailGenerator.Request(
+                        fileAt: url.standardizedFileURL,
+                        size: newSize,
+                        scale: scale,
+                        representationTypes: .thumbnail
+                    )
+                    generator.generateBestRepresentation(for: request) { thumbnail, error in
+                        if let image = thumbnail?.uiImage {
+                            let cropped = image.resized(to: size)
+                            completion(cropped)
+                        } else if error != nil {
+                            // TODO: need to handle error state
+                        }
+                    }
+                }
+            )
+        } else {
+            return .init { _, _, completion in
```

Comment:
In case we if need this `else` case, can we think of fallback, like provide laggy but non-crashing solution?

Reply:
It's not necessary because it will be removed once deployment target is changed.